### ISSUE TITLE
Copy command returns generic 404 instead of a NoSuchKey error

### ISSUE THREAD

**Issue by k2r79** [adobe/S3Mock#459]
**Exception**

When trying to copy a non-existing S3 object to the same bucket but with a different key a generic 404 exception occurs.

```
Exception in thread "main" software.amazon.awssdk.services.s3.model.S3Exception: null (Service: S3, Status Code: 404, Request ID: null, Extended Request ID: null)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handleErrorResponse(AwsXmlPredicatedResponseHandler.java:156)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handleResponse(AwsXmlPredicatedResponseHandler.java:106)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handle(AwsXmlPredicatedResponseHandler.java:84)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handle(AwsXmlPredicatedResponseHandler.java:42)
```

**Reproducing  the issue**

I use the following scratch file :

```java
import software.amazon.awssdk.services.s3.S3Client;
import software.amazon.awssdk.services.s3.model.CopyObjectRequest;

import java.io.IOException;
import java.net.URI;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;

class S3Test {

    public static void main(String[] args) throws IOException {
        final S3Client s3Client = S3Client.builder()
                                          .endpointOverride(URI.create("http://localhost:9090"))
                                          .build();

        final String copySource = URLEncoder.encode("my-bucket/test-file.json", StandardCharsets.UTF_8.toString());

        s3Client.copyObject(CopyObjectRequest.builder()
                                             .copySource(copySource)
                                             .destinationBucket("my-bucket")
                                             .destinationKey("test-file-copy.json")
                                             .build());
    }
}
```

Please note I set this environment variable to initialize the bucket : *initialBuckets=my-bucket*

**Possible fix**

A quick fix I found was to change `FileStore`'s line 701 :

```java
  // ...
  public CopyObjectResult copyS3ObjectEncrypted(final String sourceBucketName,
      final String sourceObjectName,
      final String destinationBucketName,
      final String destinationObjectName,
      final String encryption,
      final String kmsKeyId,
      final Map<String, String> userMetadata) throws IOException {
    final S3Object sourceObject = getS3Object(sourceBucketName, sourceObjectName);
    if (sourceObject == null) {
      //return null;
      throw new S3Exception(NOT_FOUND.value(), "NoSuchKey", "The specified key does not exist.");
    }
    // ...
```

I can raise a Pull Request if needed 😊

**Comment by afranken** [comment]
Thanks for the report.
S3Mock must verify that the source exists before trying to copy data, see #462. :)

**Comment by afranken** [comment]
@k2r79 just released 2.4.9 containing this fix. :)
